### ISSUE TITLE
[MD/SNES] Hiding overscan in games cuts off too much of the screen

### ISSUE THREAD

**Issue by emilos260** [higan-emu/ares#1675]
After disabling the "overscan" option in the video settings, I noticed that every Mega Drive and SNES game was cut off at the bottom of the screen. You can see it clearly in the screenshots below. In Final Fight on the SNES the bottom licensing text is cut off and in Ultimate Mortal Kombat 3 the intro text is cut off also. Is this how it's supposed to work? If so, is it possible to lower the amount of pixels "cut off" by the overscan option? 

1. Go to 'Settings', then to 'Video'
2. Uncheck 'Overscan' under 'Emulator settings' section
3. See the problem

**Screenshots**

![Final Fight - Arcade Remix (Hack) v1 1 RealGaea 2024-10-24 21-35-09](https://github.com/user-attachments/assets/86ef7ee8-93d4-4c2c-a204-8ca35f130116)
![Final Fight - Arcade Remix (Hack) v1 1 RealGaea 2024-10-24 21-35-36](https://github.com/user-attachments/assets/73bf0a20-2ce6-4c05-ba83-50ed1105bd68)

![UMK3 Arcade Restore (Hack) v0 6 Nemesis_c 2024-10-24 21-37-17](https://github.com/user-attachments/assets/aab6a5cf-cfd9-4645-8f7a-ae59490b372e)
![UMK3 Arcade Restore (Hack) v0 6 Nemesis_c 2024-10-24 21-37-22](https://github.com/user-attachments/assets/f8a75906-9a1d-47fa-ad14-ce7c43ac8f2a)

**Comment by TwentySixHundred2600** [comment]
I have the same issue and from what i can see is it crops about 2-4 pixels too much on both top and bottom. I am thinking it would be something to do with the way it scales the frame.

**Comment by DanKorell** [comment]
Yep, I can also confirm this. Tried v140 and the nightly build from 02-Nov-2024, and both have the issue.

**Comment by FitzRoyX** [comment]
So long story short, the common resolutions that get tossed around online are basically talking about "active display" aka the drawable region of the output signal. The full signal including undrawable areas is bigger (and emptier). This commit changed us to full signal.
https://github.com/ares-emulator/ares/commit/acaa1cabb53f3c62131189d9f5650b87298c3b4f

But the new numbers or math are off somehow with overscan disabled. Sonic 1 should be displaying at 320x224. Instead it's displaying at 319x219. Similar over-chops are happening across other systems.

There's also the added complication of what to do about the padding coded WITHIN active display. The Neo Geo, for example, has 320x224 active display, but half of its games drew 304x224. The SNES is similar, active display is 256x240, but line 0 is forceblanked and a settable bit blanks another 15 lines out.
https://neogeo-megashock.blogspot.com/p/neo-geo-resolution.html
https://forums.nesdev.org/viewtopic.php?p=112837#p112837

Back when active display was all we cared about, disabling "overscan" meant cutting off this "active padding". Since no one cared about seeing the dead space of a full signal, maybe all that really needed to be done was to rename "overscan" to something like "show full active display for NG and SNES". Though if you read that neo geo link, you might say screw it. It's much more straightforward to just display all of active display than try to perfectly cut off 10 different permutations of active display padding.

**Comment by FitzRoyX** [comment]
Overscan on
1415x486
![1](https://github.com/user-attachments/assets/eb1a7189-2d93-469c-b30f-d43c3e6ad4f3)

Overscan off
1280x448
![2](https://github.com/user-attachments/assets/667423d6-d2bb-4501-8055-62c37297b442)

See the truncation on each side?

**Comment by LukeUsher** [comment]
adjusted in [fa1c21b](https://github.com/ares-emulator/ares/commit/fa1c21ba3d3434495fad20312d6a8ea172a598ec)

![Image](https://github.com/user-attachments/assets/64eb6ebf-8fc5-4c27-9d6a-8f2c7956fe18)

**Comment by remutro** [comment]
Fixed in Tasco's PR (for MD games using V30), and SNES fixed by FitzRoyX's PR. 
Available in latest nightly build for now.
